lib/ohai/plugins/network.rb

```diff
@@ -25,6 +25,13 @@
 
   depends "network/interfaces"
 
+  # Try to u32 an IPv4 and fallback to u128 if it fails before throwing
+  def int_an_ip(ipaddress)
+    ipaddress.to_u32
+  rescue NoMethodError
+    ipaddress.to_u128
+  end
```

Comment:
Cathing NoMethodError leads to all sorts of fun bugs. Instead try:

```
ipaddress.respond_to?(:to_u32) ? ipaddress.to_u32 : ipaddress.to_u128
```

Reply:
Actually, why are we doing this at all? `.to_i` method does the right thing in all cases:

```
irb(main):003:0> x = IPAddress('10.1.1.1')
=> #<IPAddress::IPv4:0x0000000002aa1930 @address="10.1.1.1", @prefix=32, @octets=[10, 1, 1, 1], @u32=167837953>
irb(main):004:0> x.ipv4?
=> true
irb(main):005:0> x.to_u32
=> 167837953
irb(main):006:0> x.to_i
=> 167837953
irb(main):007:0> x = IPAddress('fe80::96c6:91ff:fea0:7a03')
=> #<IPAddress::IPv6:0x00000000024ad678 @groups=[65152, 0, 0, 0, 38598, 37375, 65184, 31235], @address="fe80:0000:0000:0000:96c6:91ff:fea0:7a03", @compressed="fe80::96c6:91ff:fea0:7a03", @prefix=128>
irb(main):008:0> x.to_i
=> 338288524927261089664883428521100212739
irb(main):009:0> x.to_u128
=> 338288524927261089664883428521100212739
```